examples/document.yaml

```diff
@@ -0,0 +1,43 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+on:
+  push:
+    paths: ["R/**"]
+  pull_request:
+    paths: ["R/**"]
+
+name: Document
+
+jobs:
+  document:
```

Comment:
AFAICT a pull request may steal the token or do other malicious actions. I think you need to restrict this action to members and owners, like this: https://github.com/r-lib/actions/blob/b03bfcfd29726270488b596e0b3dc2bafc9cae0b/examples/pr-commands.yaml#L11

Reply:
Thanks for bringing this up—it was something I had not considered. I did a bit more digging into this and, while I suspect that it is actually fine to leave as is, I see no harm in adding something similar. Reading some of Githubs' docs, they actually discuss this in some detail [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/). To quote them,

> Due to the dangers inherent to automatic processing of PRs, GitHub’s standard pull_request workflow trigger by default prevents write permissions and secrets access to the target repository.

They also write [here](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories)

> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository. The permissions for the `GITHUB_TOKEN` in forked repositories is read-only.

Based on these quotes, it seems that using the `pull_request` trigger is safe against attacks (at least attempts to steal secrets).

What do you think? Adding the `if` statement might provide extra protection/sense of security...